cassandra/metadata.py

```diff
@@ -629,10 +629,14 @@ def make_token_replica_map(self, token_to_host_owner, ring):
         hosts_per_dc = defaultdict(set)
         for i, token in enumerate(ring):
             host = token_to_host_owner[token]
-            dc_to_token_offset[host.datacenter].append(i)
-            if host.datacenter and host.rack:
-                dc_racks[host.datacenter].add(host.rack)
-                hosts_per_dc[host.datacenter].add(host)
+            host_dc = host.datacenter
+            if host_dc in dc_rf_map:
```

Comment:
Even better optimization would be to have a set of hosts run this logic over it, instead of running it for all tokens.